pkg/driver/mysql.go

```diff
@@ -0,0 +1,119 @@
+package driver
+
+import (
+	"context"
+	"database/sql/driver"
+	"github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+// MySQLDriver extends mysql.MySQLDriver with auto-SETting Galera cluster options.
+type MySQLDriver struct {
+	mysql.MySQLDriver
+}
+
+// Open implements the driver.Driver interface.
+func (md MySQLDriver) Open(name string) (driver.Conn, error) {
+	connector, err := md.OpenConnector(name)
+	if err != nil {
+		return nil, err
+	}
+
+	return connector.Connect(context.Background())
+}
+
+// OpenConnector implements the driver.DriverContext interface.
+func (md MySQLDriver) OpenConnector(name string) (driver.Connector, error) {
+	var wsrepSyncWait int64
+
+	if config, err := mysql.ParseDSN(name); err == nil {
+		if s, ok := config.Params["wsrep_sync_wait"]; ok {
+			if i, err := strconv.ParseInt(s, 10, 64); err == nil {
+				// MySQL single nodes don't know wsrep_sync_wait and fail with error 1193 "Unknown system variable".
+				// We have to SET it manually later and swallow error 1193 not to fail our connections.
+				wsrepSyncWait = i
+				delete(config.Params, "wsrep_sync_wait")
+				name = config.FormatDSN()
+			}
+		}
```

Comment:
Since we recently bumped the [`mysql`](https://github.com/Icinga/icingadb/pull/687) driver to version `1.8.0`, how about registering a before connect callback like this instead?

```go
mysql.BeforeConnect(func(ctx context.Context, config *mysql.Config) error {
    // Do some stuff with "wsrep_sync_wait"
})
```

That way we wouldn't have to parse the DSN twice for each connection.

Reply:
This doesn't make any difference to the current version.